src/partial.rs

```diff
@@ -0,0 +1,50 @@
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+use crate::annotate::{IsDeserializer, IsSerializer};
+use crate::Deserializer as AnnotatedDeserializer;
+use crate::{Document, Error};
+
+impl Serialize for Document {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        if serializer.is_serde_annotate() {
+            unsafe {
+                // If `serializer` is the correct type, then we can clone the
+                // Document node and return it.
+                let r: Result<Document, Error> = Ok(self.clone());
```

Comment:
Should probably move this line out of the `unsafe` block, because unless I am missing something it doesn't require `unsafe`.

Reply:
Only the transmute needs unsafe.  Fixed.